### DISCUSSION TITLE
Codeblock Diffing Capabilities

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2760] [Q&A] [answered]
Hi Paste!

Currently in Evergreen, we have a Code Block that has capabilities to differentiate what the changes are. This is important to Segment users to understand the impact of their changes. This also supplements a few revision history patterns we have. We also have line numbers inside the code blocks.

![CleanShot 2022-10-21 at 11 09 20](https://user-images.githubusercontent.com/1590723/197228734-5d4671a7-8a54-4810-a73a-4abbb136ce75.png)

Can you provide us with more information on how we should approach this component? Can the Paste [Code Block](https://paste.twilio.design/components/code-block) somehow be configured to include this? Or is this a component Segment should build for our purposes?

**Comment by gloriliale** [comment, marked as answer]
Hi @brannytempy! 

Though the Code Block component does include the option for line numbers (the `showLineNumbers` prop), the syntax highlighting library we use would only be able to highlight diffs by line, not character, which is what your examples show. I think in this case, it would make more sense for Segment to build a custom component to get the specific functionality you're looking for!
